### DISCUSSION TITLE
A couple of questions based off the chainlink VRF.

### DISCUSSION THREAD

**Discussion by mxnwafor** [Cyfrin/foundry-full-course-cu#1803] [Q&A] [answered]
1. What does the VRFCoordinator do?
2. Is the consumer the contract that wants to use the chainlink VRF?
3. What does it mean to fund a subscription? If you've already created one, why fund it? If you don't have a subscription create one.

I'm quite confused concerning these.
Thanks.

**Comment by EngrPips** [comment, marked as answer]
Hello @0xMitchy, Let's try to clarify what is confusing.

**What does the VRFCoordinator do**
The VRFCoordinator is the chainlink contract that we make a request to for a random number

**Is the consumer the contract that wants to use the chainlink VRF**
The consumer contract is a contract that every contract that wants to request a random number from the VRFCoordinator must inherit to receive the random number they request from the VRFCoordinator.

**What does it mean to fund a subscription?**
Whenever you send a request to the VRFCoordinator for a random number, you are asking the contract to do some computation to generate for you a unique and genuine random number, That is some work that someone has to pay for, So funding your subscription would make it so your subscription would cover the charges to generate that unique and genuine random number


Please let us know if anything is still confusing.

**Reply by mxnwafor** [reply]
ahh...(in Patrick's voice).
I think I understand now. Thank you @EngrPips !

**Reply by EngrPips** [reply]
Awesome, I am glad you understand it now. Keep crushing it.
